src/odemis/acq/align/z_localization.py

```diff
@@ -0,0 +1,193 @@
+    if (fit_tol / 0.98) * sigma_x < pcov[2] or (fit_tol / 0.98) * sigma_y < pcov[3]:
+        logging.warning("The image contains too much noise to determine the size of the feature accurately, "
+                        "the results may be inaccurate.\n"
+                        "The standard deviation error for determined feature size in x is: %s and for y is %s.",
+                        pcov[2], pcov[3])
+        warning = 5
+    elif sigma_x > 0.85 * image.shape[0] or sigma_y > 0.85 * image.shape[1]:
+        logging.warning("The detected size of the feature in the image is too big w.r.t. the size of the image."
+                        "Using an image with more space around the feature might solve this problem."
+                        "The detected feature size is %s in x and %s in y direction, while the total image has a "
+                        "width of %s pixels and a height of %s pixels."
+                        "The feature size should be < 85% in each direction\n"
+                        "Current results may be inaccurate.", sigma_x, sigma_y, image.shape[0], image.shape[1])
+        warning = 6
+
+    # Determine the z position using the shape of the features (sigma_x/sigma_y)
+    max_range = calibration_data["z_calibration_range"][1] - calibration_data["z_calibration_range"][0]
+    fine_z = numpy.arange(0, max_range, z_stack_step_size*1e9)
+    est_func = huang(fine_z, calibration_data["x"]) - huang(fine_z, calibration_data["y"])
+    x0 = fine_z[numpy.abs(est_func - (sigma_x - sigma_y)).argmin()]  # A raw initial estimate of the z position
+
+    # Apply the up sample factor just as done in the calibration
+    sigma_x *= calibration_data['upsample_factor']
+    sigma_y *= calibration_data['upsample_factor']
+    zopt, _, _, _, warn_flag, _ = fmin_cg(solve_psf,
+                                          x0=x0,
+                                          args=(sigma_x, sigma_y, calibration_data, huang),
+                                          maxiter=2000, disp=False, full_output=True,
+                                          retall=True,
+                                          )
+
+    if warn_flag > 0:
+        logging.warning(f"Inaccuracy observed during when determining the Z position, the warning flag {warn_flag}"
+                        f"was raised. Current results may be inaccurate")
+        # fmin_cg output the warning flags 1, 2 and 3. The warnings of fmin_cg may be te result of previously found
+        # warning, therefore these are considered more important
+        if not(warning):
+            warning = 4
+
+    z_position = round(zopt[0]*1e-9 - calibration_data["z_least_confusion"], 20)
+
+    if not(calibration_data["z_calibration_range"][0] < z_position < calibration_data["z_calibration_range"][1]):
+        # Always log this warning but only update this error if no other cause of the error is found.
+        logging.warning("The determined z position is out of the specified max_range."
```

Comment:
Please log the z position next to "z position" and  the "max_range" value next to "max_range". It'll be easier to debug.